### PR TITLE
Some wamr-ide improvements

### PR DESCRIPTION
**Pull request by yamt**
including:
* macOS support
* documentation improvements
* fix warnings

### CODE REVIEW

File: test-tools/wamr-ide/WASM-Debug-Server/Docker/README.md [hunk 1]
```diff
@@ -3,7 +3,6 @@
 -   Linux
 
     ```shell
-    chmod +x resource/*
```
Comment (xujuntwt95329) [inline test-tools/wamr-ide/WASM-Debug-Server/Docker/README.md:6]: Sometimes the files may lost executable permissions (e.g. copied from windows), had better add a notice that 
> Note: Ensure the scripts under `resource/scripts` and binaries under `resource/debug/bin` has executable permission

File: test-tools/wamr-ide/README.md [hunk 2]
```diff
@@ -33,14 +33,14 @@ The WAMR-IDE is an Integrated Development Environment to develop WebAssembly app
...
-We have 2 docker images which should be built or loaded on your host, `wasm-toolchain` and `wasm-debug-server`. To build these 2 images, please enter the `WASM_Source_Debug_Server/Docker` & `WASM_Toolchain/Docker`, then execute the `build_docker_image` script respectively.
+We have 2 docker images which should be built or loaded on your host, `wasm-toolchain` and `wasm-debug-server`. To build these 2 images, please enter the `WASM-Debug-Server/Docker` & `WASM-Toolchain/Docker`, then execute the `build_docker_image` script respectively.
 
 Windows (powershell):
 
 ```batch
 $ cd .\WASM-Toolchain\Docker
 $ ./build_docker_image.bat
-$ cd .\WASM-Source-Debug-Server\Docker
+$ cd .\WASM-Debug-Server\Docker
```
Comment (wenyongh) [inline test-tools/wamr-ide/README.md:43]: L42 and L44, we had better change `./` to `.\` as it is on Windows platform. Could you please help change them?

File: test-tools/wamr-ide/VSCode-Extension/resource/scripts/boot_debugger_server.sh [hunk 3]
```diff
@@ -1,13 +1,13 @@
...
 docker run -it --name=wasm-debug-server-ctr \
            -v "$(pwd)":/mnt \
            -p 1234:1234 \
            wasm-debug-server:1.0 \
            /bin/bash -c "./debug.sh $1"
 
 docker stop wasm-debug-server-ctr>/dev/null
-docker rm wasm-debug-server-ctr>/dev/null
\ No newline at end of file
+docker rm wasm-debug-server-ctr>/dev/null
```
Comment (wenyongh) [inline test-tools/wamr-ide/VSCode-Extension/resource/scripts/boot_debugger_server.sh:13]: Shall we use `docker rm -f` like the command modified in other files?

File: test-tools/wamr-ide/VSCode-Extension/resource/scripts/build.sh [hunk 4]
```diff
@@ -1,13 +1,13 @@
...
 docker run --name=wasm-toolchain-ctr \
                 -it -v "$(pwd)":/mnt \
                 --env=PROJ_PATH="$(pwd)" \
                 wasm-toolchain:1.0  \
                 /bin/bash -c "./build_wasm.sh $1"
 
 docker stop wasm-toolchain-ctr>/dev/null
-docker rm wasm-toolchain-ctr>/dev/null
\ No newline at end of file
+docker rm wasm-toolchain-ctr>/dev/null
```
Comment (wenyongh) [inline test-tools/wamr-ide/VSCode-Extension/resource/scripts/build.sh:13]: Same as above

File: test-tools/wamr-ide/WASM-Debug-Server/Docker/README.md (hunk 1 above)
Comment (yamt) [reply test-tools/wamr-ide/WASM-Debug-Server/Docker/README.md:6]: ok. i added a note.

File: test-tools/wamr-ide/README.md (hunk 2 above)
Comment (yamt) [reply test-tools/wamr-ide/README.md:49]: done

File: test-tools/wamr-ide/VSCode-Extension/resource/scripts/boot_debugger_server.sh (hunk 3 above)
Comment (yamt) [reply test-tools/wamr-ide/VSCode-Extension/resource/scripts/boot_debugger_server.sh]: i simplified them differently.

File: test-tools/wamr-ide/VSCode-Extension/resource/scripts/build.sh (hunk 4 above)
Comment (yamt) [reply test-tools/wamr-ide/VSCode-Extension/resource/scripts/build.sh]: ditto

File: test-tools/wamr-ide/README.md [hunk 5]
```diff
@@ -89,9 +95,21 @@ $ docker build --no-cache --build-arg http_proxy=http://proxy-example.com:1234
...
-> **You can also debug the extension directly follow this [instruction](./VSCode_Extension/README.md) without packing the extension.**
+You may need to install typescript as well.
+
+```shell
+$ npm install -g vsce
+$ npm install -g typescript
+$ cd VSCode-Extension
+$ vsce package
+```
+
```
Comment (NingW101) [inline test-tools/wamr-ide/README.md:106]:
```suggestion

```shell
$ npm install -g vsce
$ cd VSCode-Extension
$ rm -fr node_modules
$ npm install
$ vsce package
```

```

File: test-tools/wamr-ide/WASM-Toolchain/Docker/README.md [hunk 6]
```diff
@@ -3,7 +3,6 @@
 -   Linux
 
     ```shell
-    chmod +x resource/*
```
Comment (NingW101) [inline test-tools/wamr-ide/WASM-Toolchain/Docker/README.md:6]: Same as above

File: test-tools/wamr-ide/README.md (hunk 5 above)
Comment (yamt) [reply test-tools/wamr-ide/README.md:106]: ok. thank you. done.

File: test-tools/wamr-ide/WASM-Toolchain/Docker/README.md (hunk 6 above)
Comment (yamt) [reply test-tools/wamr-ide/WASM-Toolchain/Docker/README.md:6]: i added a note in wamr-ide/README.md

**Pull request merged**